In [1]:
import numpy as no
import pandas as pd
import os
from pathlib import Path
import swifter

In [2]:
os.getcwd()

'd:\\GitHub\\MSBD5005_DataVizProject'

In [3]:
base_path = (Path(os.getcwd()) / 'Datasets/Activity Logs').resolve()
all_activity_log_file_name = os.listdir(base_path)

In [4]:
raw_file = []
for file in all_activity_log_file_name:
    if '.csv' in file:
        print(file)
        raw_file.append(pd.read_csv(base_path / file))

ParticipantStatusLogs1.csv
ParticipantStatusLogs10.csv
ParticipantStatusLogs11.csv
ParticipantStatusLogs12.csv
ParticipantStatusLogs13.csv
ParticipantStatusLogs14.csv
ParticipantStatusLogs15.csv
ParticipantStatusLogs16.csv
ParticipantStatusLogs17.csv
ParticipantStatusLogs18.csv
ParticipantStatusLogs19.csv
ParticipantStatusLogs2.csv
ParticipantStatusLogs20.csv
ParticipantStatusLogs21.csv
ParticipantStatusLogs22.csv
ParticipantStatusLogs23.csv
ParticipantStatusLogs24.csv
ParticipantStatusLogs25.csv
ParticipantStatusLogs26.csv
ParticipantStatusLogs27.csv
ParticipantStatusLogs28.csv
ParticipantStatusLogs29.csv
ParticipantStatusLogs3.csv
ParticipantStatusLogs30.csv
ParticipantStatusLogs31.csv
ParticipantStatusLogs32.csv
ParticipantStatusLogs33.csv
ParticipantStatusLogs34.csv
ParticipantStatusLogs35.csv
ParticipantStatusLogs36.csv
ParticipantStatusLogs37.csv
ParticipantStatusLogs38.csv
ParticipantStatusLogs39.csv
ParticipantStatusLogs4.csv
ParticipantStatusLogs40.csv
ParticipantStatusLogs41.

In [5]:
raw_file = pd.concat(raw_file, axis = 0)


In [6]:
raw_file.shape

(113923735, 12)

In [7]:
raw_file = raw_file.reset_index(drop=True)

In [8]:
# type(raw_file['timestamp'][0])
raw_file['timestamp'] = pd.to_datetime( raw_file['timestamp'], format="%Y-%m-%dT%H:%M:%SZ")

In [9]:
raw_file.head()

timestamp                                 currentLocation  participantId  \
0 2022-03-01  POINT (-2724.6277665310454 6866.2081834436985)            0.0   
1 2022-03-01  POINT (-1526.9372331431534 5582.2951345645315)            1.0   
2 2022-03-01   POINT (-1360.9905987829304 2108.804385379679)            2.0   
3 2022-03-01    POINT (-1558.517200825967 5600.664347152427)            3.0   
4 2022-03-01     POINT (976.2409614204214 4574.575079082071)            4.0   

  currentMode hungerStatus sleepStatus  apartmentId  availableBalance  jobId  \
0      AtHome      JustAte    Sleeping        926.0       1286.519556  254.0   
1      AtHome      JustAte    Sleeping        928.0        860.574204  929.0   
2      AtHome      JustAte    Sleeping        291.0       1298.184541  348.0   
3      AtHome      JustAte    Sleeping       1243.0       1180.641725  316.0   
4      AtHome      JustAte    Sleeping        194.0       -681.650588  177.0   

  financialStatus  dailyFoodBudget  weeklyExtraBudget  
0          Stable             12.0        1104.302570  
1          Stable             12.0         926.714377  
2          Stable             16.0         848.802876  
3          Stable             12.0         819.325405  
4        Unstable             20.0           0.000000

In [10]:
raw_file['date'] = raw_file['timestamp'].dt.date
# raw_file.iloc[:3,:]['timestamp'].dt.date

In [18]:
daily_pid_jid = raw_file[['date', 'participantId', 'jobId']].drop_duplicates()
daily_pid_jid['date'].value_counts().plot(kind ='line')

In [38]:
daily_jid = daily_pid_jid[['date', 'jobId']]

In [48]:
daily_jid = daily_jid.merge(right=job_df[['jobId', 'employerId']], on=['jobId'], how='left')

In [127]:
daily_jid.head()

date  jobId  employerId
0  2022-03-01  254.0       424.0
1  2022-03-01  929.0      1321.0
2  2022-03-01  348.0       440.0
3  2022-03-01  316.0       436.0
4  2022-03-01  177.0       411.0

In [79]:
att_base_path = base_path.parent / 'Attributes'
job_df = pd.read_csv(att_base_path / 'Jobs.csv')
employer_df = pd.read_csv(att_base_path / 'Employers.csv')

There are toally 1192 Jobs Id and 254 Employers. 

Job ID range: [1 .. 1326] (1325)

Employer ID range: [379 .. 1797] (1419)

In [92]:
pubs_df = pd.read_csv(att_base_path / 'Pubs.csv')
food_df = pd.read_csv(att_base_path / 'Restaurants.csv')
school_df = pd.read_csv(att_base_path / 'Schools.csv')

In [96]:
pubs_df['type'] = 'Pub'
pubs_df = pubs_df[['pubId', 'buildingId', 'type']]
pubs_df.columns = ['id', 'buildingId', 'type']

In [97]:
food_df['type'] = 'Restaurant'
food_df = food_df[['restaurantId', 'buildingId', 'type']]
food_df.columns = ['id', 'buildingId', 'type']

In [100]:
school_df['type'] = 'School'
school_df = school_df[['schoolId', 'buildingId', 'type']]
school_df.columns = ['id', 'buildingId', 'type']


In [102]:
workplace = pd.concat([pubs_df, food_df, school_df], axis = 0)

In [105]:
workplace = workplace.sort_values(by=['buildingId'])

In [106]:
employer_df[['employerId', 'buildingId']].mrege(right=workplace, o)

employerId                                       location  buildingId
0           379  POINT (-1849.997168394888 1744.6010147106394)         823
1           380    POINT (41.51783767879146 418.7264799744545)         154
2           381   POINT (877.2786575380362 1358.5441805909259)         279
3           382   POINT (670.3987400004884 1584.4743462106067)           3
4           383   POINT (829.9556783260775 2163.4803049897623)         146
..          ...                                            ...         ...
248        1793  POINT (-3313.211912858075 6185.8337275541235)         738
249        1794   POINT (-3786.373075235378 4999.485452069126)         802
250        1795    POINT (-3267.606054814424 6452.56705940432)         953
251        1796   POINT (-3725.261225376805 6345.551342169127)        1011
252        1797   POINT (-3759.480918044369 5179.602735764589)         949

[253 rows x 3 columns]

In [113]:
workplace

id  buildingId        type
11  1346          27  Restaurant
1    443          29         Pub
2    447          58  Restaurant
9    899          86  Restaurant
18  1804         101  Restaurant
3   1350         123      School
13  1348         124  Restaurant
15  1801         160  Restaurant
5    895         164  Restaurant
4    893         164         Pub
4    449         181  Restaurant
17  1803         213  Restaurant
10  1799         234         Pub
5    894         238         Pub
2    900         262      School
19  1805         285  Restaurant
0    445         304  Restaurant
1    446         308  Restaurant
6   1342         429         Pub
7   1343         489         Pub
3    892         502         Pub
0    442         556         Pub
8   1344         585         Pub
6    896         619  Restaurant
11  1800         627         Pub
0      0         662      School
12  1347         679  Restaurant
16  1802         714  Restaurant
7    897         875  Restaurant
14  1349         888  Restaurant
8    898         917  Restaurant
1    450         943      School
9   1798         953         Pub
3    448         964  Restaurant
10  1345         991  Restaurant
2    444        1012         Pub

In [125]:
employer_df.merge(workplace, on = 'buildingId', how ='left')

employerId                                       location  buildingId  \
0           379  POINT (-1849.997168394888 1744.6010147106394)         823   
1           380    POINT (41.51783767879146 418.7264799744545)         154   
2           381   POINT (877.2786575380362 1358.5441805909259)         279   
3           382   POINT (670.3987400004884 1584.4743462106067)           3   
4           383   POINT (829.9556783260775 2163.4803049897623)         146   
..          ...                                            ...         ...   
248        1793  POINT (-3313.211912858075 6185.8337275541235)         738   
249        1794   POINT (-3786.373075235378 4999.485452069126)         802   
250        1795    POINT (-3267.606054814424 6452.56705940432)         953   
251        1796   POINT (-3725.261225376805 6345.551342169127)        1011   
252        1797   POINT (-3759.480918044369 5179.602735764589)         949   

         id type  
0       NaN  NaN  
1       NaN  NaN  
2       NaN  NaN  
3       NaN  NaN  
4       NaN  NaN  
..      ...  ...  
248     NaN  NaN  
249     NaN  NaN  
250  1798.0  Pub  
251     NaN  NaN  
252     NaN  NaN  

[253 rows x 5 columns]

In [ ]:
import ast

In [131]:
daily_jid.groupby(by=['date', 'employerId']).size().reset_index(drop=False).to_csv('total_employee_for_each_job.csv')

In [164]:
job_df['endTime'] = pd.to_datetime(job_df['endTime'], format='%H:%M:%S %p')

In [165]:
job_df

jobId  employerId  hourlyRate           startTime             endTime  \
0         0         379   10.000000 1900-01-01 07:46:00 1900-01-01 03:46:00   
1         1         379   22.217633 1900-01-01 07:31:00 1900-01-01 03:31:00   
2         2         380   10.000000 1900-01-01 08:00:00 1900-01-01 04:00:00   
3         3         380   15.312071 1900-01-01 07:39:00 1900-01-01 03:39:00   
4         4         381   21.355409 1900-01-01 07:53:00 1900-01-01 03:53:00   
...     ...         ...         ...                 ...                 ...   
1323   1323        1796   37.166553 1900-01-01 07:29:00 1900-01-01 03:29:00   
1324   1324        1796   23.815211 1900-01-01 08:14:00 1900-01-01 04:14:00   
1325   1325        1797   13.117569 1900-01-01 07:37:00 1900-01-01 03:37:00   
1326   1326        1797   10.000000 1900-01-01 07:31:00 1900-01-01 03:31:00   
1327   1327        1797   11.932836 1900-01-01 08:19:00 1900-01-01 04:19:00   

                                       daysToWork educationRequirement  \
0      [Monday,Tuesday,Wednesday,Thursday,Friday]  HighSchoolOrCollege   
1      [Monday,Tuesday,Wednesday,Thursday,Friday]            Bachelors   
2      [Monday,Tuesday,Wednesday,Thursday,Friday]            Bachelors   
3      [Monday,Tuesday,Wednesday,Thursday,Friday]            Bachelors   
4      [Monday,Tuesday,Wednesday,Thursday,Friday]  HighSchoolOrCollege   
...                                           ...                  ...   
1323   [Monday,Tuesday,Wednesday,Thursday,Friday]             Graduate   
1324  [Friday,Saturday,Thursday,Monday,Wednesday]            Bachelors   
1325   [Monday,Tuesday,Wednesday,Thursday,Friday]  HighSchoolOrCollege   
1326   [Monday,Tuesday,Wednesday,Thursday,Friday]  HighSchoolOrCollege   
1327   [Monday,Tuesday,Wednesday,Thursday,Friday]            Bachelors   

      days_per_week  monthly_salary  
0                 5      400.000000  
1                 5      888.705334  
2                 5      400.000000  
3                 5      612.482826  
4                 5      854.216372  
...             ...             ...  
1323              5     1486.662121  
1324              5      952.608439  
1325              5      524.702758  
1326              5      400.000000  
1327              5      477.313431  

[1328 rows x 9 columns]

In [156]:
# job_df['daysToWork'] = job_df['daysToWork'].apply(lambda x: ast.literal_eval(x))
job_df['days_per_week'] = job_df['daysToWork'].apply(lambda x: len(x.split(',')))

In [173]:
job_df['monthly_salary'] = job_df[['hourlyRate', 'days_per_week']].apply(lambda x: float(x[0]) * x[1] * 8 * 4 , axis=1)

In [ ]:
job_df['monthly_salary'] = job_df[['hourlyRate', 'days_per_week']].apply(lambda x: float(x[0]) * x[1] * 8, axis=1)

In [175]:
# job_df[['jobId', 'monthly_salary']]
daily_jid = daily_jid.merge(job_df[['jobId', 'monthly_salary']], on=['jobId'], how='left')

In [188]:
daily_jid['yr_mn'] = daily_jid['date'].apply(lambda x: str(x.year) + '-' + str(x.month))

In [190]:
daily_jid

date   jobId  employerId  monthly_salary   yr_mn
0       2022-03-01   254.0       424.0     8225.600056  2022-3
1       2022-03-01   929.0      1321.0     7228.227325  2022-3
2       2022-03-01   348.0       440.0     6334.912315  2022-3
3       2022-03-01   316.0       436.0     6122.812550  2022-3
4       2022-03-01   177.0       411.0     7856.429739  2022-3
...            ...     ...         ...             ...     ...
398713  2022-04-25  1047.0      1739.0     1600.000000  2022-4
398714  2022-04-25  1226.0      1776.0     1600.000000  2022-4
398715  2022-04-25    55.0       389.0     3251.477298  2022-4
398716  2022-04-25   627.0       877.0     1600.000000  2022-4
398717  2022-04-25   143.0       406.0     1930.824045  2022-4

[398718 rows x 5 columns]

In [194]:

daily_pid_jid = daily_pid_jid.merge(job_df[['jobId', 'employerId', 'monthly_salary']], on ='jobId', how='left')

In [196]:
daily_pid_jid['yrmn'] = daily_pid_jid['date'].apply(lambda x: str(x.year) + '-' + str(x.month))

In [201]:
employer_monthly_wages_to_employee = daily_pid_jid[['employerId', 'participantId', 'monthly_salary', 'yrmn']].drop_duplicates().groupby(by=['employerId', 'yrmn'])['monthly_salary'].sum().reset_index(drop=False)

In [203]:
employer_monthly_wages_to_employee.to_csv('employer_monthly_wages_to_employee.csv')

In [205]:
jornal_base_path = base_path.parent / 'Journals'

In [206]:
checkin = pd.read_csv(jornal_base_path / 'CheckinJournal.csv')

In [209]:
travel_df = pd.read_csv(jornal_base_path / 'TravelJournal.csv')

In [213]:
work_travel_df = travel_df[travel_df['purpose'] == 'Work/Home Commute']

In [225]:
work_travel_df['checkin_datime'] = pd.to_datetime(work_travel_df['checkInTime'], format='%Y-%m-%dT%H:%M:%SZ')

C:\Users\TANGGA\AppData\Local\Temp\ipykernel_7988\2264465874.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  work_travel_df['checkin_datime'] = pd.to_datetime(work_travel_df['checkInTime'], format='%Y-%m-%dT%H:%M:%SZ')


In [226]:
work_travel_df

participantId       travelStartTime  travelStartLocationId  \
27                1008  2022-03-01T05:45:00Z                  525.0   
33                 833  2022-03-01T06:25:00Z                  552.0   
34                 342  2022-03-01T06:15:00Z                  737.0   
36                 538  2022-03-01T06:20:00Z                 1218.0   
37                 604  2022-03-01T06:45:00Z                    NaN   
...                ...                   ...                    ...   
2099606            436  2023-05-24T16:50:00Z                  868.0   
2099610            320  2023-05-24T17:15:00Z                  855.0   
2099641            151  2023-05-24T16:15:00Z                  384.0   
2099642            239  2023-05-24T16:55:00Z                  412.0   
2099646             21  2023-05-24T16:05:00Z                  434.0   

                travelEndTime  travelEndLocationId            purpose  \
27       2022-03-01T07:00:00Z                  389  Work/Home Commute   
33       2022-03-01T07:05:00Z                  421  Work/Home Commute   
34       2022-03-01T07:05:00Z                  883  Work/Home Commute   
36       2022-03-01T07:05:00Z                 1289  Work/Home Commute   
37       2022-03-01T07:05:00Z                 1282  Work/Home Commute   
...                       ...                  ...                ...   
2099606  2023-05-24T17:00:00Z                  570  Work/Home Commute   
2099610  2023-05-24T17:20:00Z                  826  Work/Home Commute   
2099641  2023-05-24T16:50:00Z                 1157  Work/Home Commute   
2099642  2023-05-24T17:00:00Z                   83  Work/Home Commute   
2099646  2023-05-24T16:20:00Z                  169  Work/Home Commute   

                  checkInTime          checkOutTime  startingBalance  \
27       2022-03-01T07:00:00Z  2022-03-01T07:05:00Z      2681.567578   
33       2022-03-01T07:05:00Z  2022-03-01T07:10:00Z       834.891712   
34       2022-03-01T07:05:00Z  2022-03-01T07:10:00Z       745.986911   
36       2022-03-01T07:05:00Z  2022-03-01T07:10:00Z       831.992210   
37       2022-03-01T07:05:00Z  2022-03-01T07:10:00Z      2361.640284   
...                       ...                   ...              ...   
2099606  2023-05-24T17:00:00Z  2023-05-24T23:25:00Z      5470.660469   
2099610  2023-05-24T17:20:00Z  2023-05-24T23:30:00Z     73095.528065   
2099641  2023-05-24T16:50:00Z  2023-05-25T00:00:00Z     34884.798703   
2099642  2023-05-24T17:00:00Z  2023-05-25T00:00:00Z      2339.820693   
2099646  2023-05-24T16:20:00Z  2023-05-25T00:00:00Z      9125.593254   

         endingBalance      checkin_datime  
27         2683.261056 2022-03-01 07:00:00  
33          836.033687 2022-03-01 07:05:00  
34          747.624208 2022-03-01 07:05:00  
36          834.608980 2022-03-01 07:05:00  
37         2362.473617 2022-03-01 07:05:00  
...                ...                 ...  
2099606    5470.660469 2023-05-24 17:00:00  
2099610   73095.528065 2023-05-24 17:20:00  
2099641   34876.798703 2023-05-24 16:50:00  
2099642    2335.820693 2023-05-24 17:00:00  
2099646    9121.593254 2023-05-24 16:20:00  

[566989 rows x 11 columns]

In [230]:
daily_pid_jid[['participantId', 'jobId']].drop_duplicates().groupby("participantId").size().reset_index(drop=False).sort_values(by=[0])

participantId   0
0              0.0   1
630          630.0   1
631          631.0   1
632          632.0   1
633          633.0   1
..             ...  ..
262          262.0  28
589          589.0  29
298          298.0  30
657          657.0  30
279          279.0  31

[1011 rows x 2 columns]

In [371]:
pid_monthly_job = daily_pid_jid[['participantId', 'jobId', 'yrmn']].groupby(by=['participantId', 'yrmn'])['jobId'].first().reset_index(drop=False).pivot(index='participantId', columns='yrmn', values='jobId')

In [374]:
pid_monthly_job.reset_index()

yrmn  participantId  2022-10  2022-11  2022-12  2022-3  2022-4  2022-5  \
0               0.0    254.0    254.0    254.0   254.0   254.0   254.0   
1               1.0    929.0    929.0    929.0   929.0   929.0   929.0   
2               2.0    348.0    348.0    348.0   348.0   348.0   348.0   
3               3.0    316.0    316.0    316.0   316.0   316.0   316.0   
4               4.0    177.0    177.0    177.0   177.0   177.0   177.0   
...             ...      ...      ...      ...     ...     ...     ...   
1006         1006.0   1047.0   1047.0   1047.0  1047.0  1047.0  1047.0   
1007         1007.0   1226.0   1226.0   1226.0  1226.0  1226.0  1226.0   
1008         1008.0     55.0     55.0     55.0    55.0    55.0    55.0   
1009         1009.0    627.0    627.0    627.0   627.0   627.0   627.0   
1010         1010.0    143.0    143.0    143.0   143.0   143.0   143.0   

yrmn  2022-6  2022-7  2022-8  2022-9  2023-1  2023-2  2023-3  2023-4  2023-5  
0      254.0   254.0   254.0   254.0   254.0   254.0   254.0   254.0   254.0  
1      929.0   929.0   929.0   929.0   929.0   929.0   929.0   929.0   929.0  
2      348.0   348.0   348.0   348.0   348.0   348.0   348.0   348.0   348.0  
3      316.0   316.0   316.0   316.0   316.0   316.0   316.0   316.0   316.0  
4      177.0   177.0   177.0   177.0   177.0   177.0   177.0   177.0   177.0  
...      ...     ...     ...     ...     ...     ...     ...     ...     ...  
1006  1047.0  1047.0  1047.0  1047.0  1047.0  1047.0  1047.0  1047.0  1047.0  
1007  1226.0  1226.0  1226.0  1226.0  1226.0  1226.0  1226.0  1226.0  1226.0  
1008    55.0    55.0    55.0    55.0    55.0    55.0    55.0    55.0    55.0  
1009   627.0   627.0   627.0   627.0   627.0   627.0   627.0   627.0   627.0  
1010   143.0   143.0   143.0   143.0   143.0   143.0   143.0   143.0   143.0  

[1011 rows x 16 columns]

{3,
 8,
 17,
 26,
 28,
 32,
 34,
 37,
 47,
 49,
 64,
 76,
 86,
 88,
 93,
 103,
 106,
 118,
 121,
 129,
 132,
 136,
 142,
 146,
 150,
 151,
 154,
 156,
 160,
 162,
 175,
 180,
 183,
 187,
 190,
 205,
 206,
 213,
 225,
 227,
 230,
 238,
 240,
 241,
 243,
 254,
 255,
 256,
 260,
 261,
 267,
 269,
 277,
 279,
 284,
 285,
 297,
 304,
 305,
 315,
 317,
 323,
 331,
 333,
 339,
 343,
 347,
 351,
 352,
 354,
 373,
 375,
 395,
 401,
 408,
 409,
 410,
 411,
 413,
 416,
 421,
 422,
 426,
 429,
 434,
 441,
 446,
 453,
 454,
 467,
 472,
 474,
 475,
 479,
 485,
 486,
 487,
 489,
 493,
 511,
 516,
 521,
 527,
 543,
 544,
 549,
 561,
 567,
 578,
 580,
 583,
 584,
 585,
 588,
 590,
 592,
 595,
 596,
 604,
 619,
 623,
 627,
 632,
 644,
 664,
 670,
 680,
 686,
 692,
 694,
 695,
 714,
 715,
 717,
 738,
 749,
 755,
 765,
 767,
 773,
 775,
 781,
 782,
 788,
 795,
 802,
 807,
 808,
 823,
 862,
 869,
 870,
 877,
 880,
 893,
 912,
 917,
 923,
 928,
 930,
 932,
 936,
 937,
 946,
 949,
 953,
 964,
 966,
 969,
 97

In [264]:
travel_df['checkin_datetime'] = pd.to_datetime(travel_df['checkInTime'], format="%Y-%m-%dT%H:%M:%SZ")

In [253]:
pid_travel_balance = travel_df[['participantId', 'travelEndLocationId', 'purpose', 'startingBalance', 'endingBalance']]
pid_travel_balance['cost'] = travel_df['']

In [256]:
pid_travel_balance['cost'] = pid_travel_balance[['startingBalance', 'endingBalance']].apply(lambda x: x[1] - x[0], axis=1)

C:\Users\TANGGA\AppData\Local\Temp\ipykernel_7988\2083086519.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  pid_travel_balance['cost'] = pid_travel_balance[['startingBalance', 'endingBalance']].apply(lambda x: x[1] - x[0], axis=1)


In [269]:
tmp = sorted(list(set(pid_travel_balance['travelEndLocationId'])))
tmp[:5] + tmp[-5:]

[0, 1, 2, 5, 9, 1801, 1802, 1803, 1804, 1805]

In [246]:
buildings = pd.read_csv(att_base_path / 'Buildings.csv')
apartment = pd.read_csv(att_base_path / 'Apartments.csv')

In [247]:
buildings

buildingId                                           location  \
0              1  POLYGON ((350.0638997002585 4595.665606173783,...   
1              2  POLYGON ((-1926.972613718425 2725.610686806701...   
2              3  POLYGON ((685.6846002015491 1552.131491805318,...   
3              4  POLYGON ((-976.7845160060303 4542.38209636188,...   
4              5  POLYGON ((1259.3061988755617 3572.726728111263...   
...          ...                                                ...   
1037        1038  POLYGON ((361.2332515102781 4376.8840047388585...   
1038        1039  POLYGON ((1011.5463728925471 362.9701764035531...   
1039        1040  POLYGON ((-3562.8074734172264 7362.92295511692...   
1040        1041  POLYGON ((-4151.699758553839 5418.613349758049...   
1041        1042  POLYGON ((-1658.924569481752 4827.536640495573...   

     buildingType  maxOccupancy                  units  
0      Commercial           NaN                    NaN  
1      Residental          12.0  [481,498,534,652,818]  
2      Commercial           NaN                  [382]  
3      Commercial           NaN                    NaN  
4      Residental           2.0                  [231]  
...           ...           ...                    ...  
1037   Residental           8.0              [118,287]  
1038   Residental           2.0                  [365]  
1039   Residental           5.0       [1406,1564,1569]  
1040   Commercial           NaN                 [1737]  
1041   Commercial           NaN                    NaN  

[1042 rows x 5 columns]

In [273]:
import numpy as np
journal_base_path = jornal_base_path

checkin_df = pd.read_csv(journal_base_path / 'CheckinJournal.csv')

In [279]:
checkin_df['datetime'] = pd.to_datetime(checkin_df['timestamp'], format='%Y-%m-%dT%H:%M:%SZ')
checkin_df['date'] = checkin_df['datetime'].dt.date

venueId   venueType        date    0
0             0   Apartment  2022-03-01  267
1             0   Apartment  2022-03-02  271
2             0   Apartment  2022-03-03  246
3             0   Apartment  2022-03-04  244
4             0   Apartment  2022-03-05    4
...         ...         ...         ...  ...
463938     1805  Restaurant  2023-05-20   88
463939     1805  Restaurant  2023-05-21   88
463940     1805  Restaurant  2023-05-22   89
463941     1805  Restaurant  2023-05-23   80
463942     1805  Restaurant  2023-05-24   85

[463943 rows x 4 columns]

In [289]:
travel_df['checkin_date'] = travel_df['checkin_datetime'].dt.date
travel_checkin_df = travel_df[['checkin_date', 'travelEndLocationId', 'purpose']].groupby(by=['checkin_date', 'travelEndLocationId', 'purpose']).size().reset_index(drop=False)
venue_daily_checkin_df = checkin_df[['venueId', 'venueType', 'date']].groupby(by=['venueId', 'venueType', 'date']).size().reset_index(drop=False)
venue_daily_checkin_df.columns =['venueId', 'venueType', 'checkin_date', 'checkin_cnt']
travel_checkin_df.columns = ['checkin_date', 'venueId', 'purpose', 'checkin_cnt_travel']
daily_venue_checkin = travel_checkin_df.merge(venue_daily_checkin_df, on=['checkin_date', 'venueId'], how='left')
daily_venue_checkin['checkin_yrwk'] = daily_venue_checkin['checkin_date'].apply(lambda x: x.strftime("%G%V"))
daily_venue_checkin['weekday'] = daily_venue_checkin['checkin_date'].apply(lambda x: x.weekday())

travel_df['cost'] = travel_df[['startingBalance', 'endingBalance']].apply(lambda x: x[0] - x[1], axis=1)
travel_df['yrwk'] = travel_df['checkin_date'].apply(lambda x: x.strftime("%G%V"))

venue_daily_spend = travel_df[['checkin_date', 'cost', 'travelEndLocationId', 'purpose']].groupby(by=['checkin_date', 'travelEndLocationId','purpose']).agg(
    income_mean=('cost', np.mean),
    income_sum=('cost', np.sum),
    income_max=('cost', np.max),
    income_min=('cost', np.min),
).reset_index(drop=False)
venue_daily_spend.columns = ['checkin_date', 'venueId', 'purpose', 'incomce_mean', 'income_sum', 'income_max', 'income_min']
daily_venue_checkin_income = daily_venue_checkin.merge(venue_daily_spend, on=['checkin_date', 'venueId', 'purpose'], how='left')
daily_venue_checkin_income['checkin_yrwk'] = daily_venue_checkin_income['checkin_date'].apply(lambda x: x.strftime('%G%V'))

In [361]:
daily_venue_checkin_income.to_csv('daily_venue_checkin_income.csv', index=False)


In [364]:
daily_venue_checkin_income

checkin_date  venueId                      purpose  checkin_cnt_travel  \
0        2022-03-01        0  Coming Back From Restaurant                 129   
1        2022-03-01        0           Going Back to Home                   7   
2        2022-03-01        0            Work/Home Commute                 131   
3        2022-03-01        1  Coming Back From Restaurant                   1   
4        2022-03-01        1           Going Back to Home                   3   
...             ...      ...                          ...                 ...   
724402   2023-05-24     1801                       Eating                 115   
724403   2023-05-24     1802                       Eating                  39   
724404   2023-05-24     1803                       Eating                  24   
724405   2023-05-24     1804                       Eating                  23   
724406   2023-05-24     1805                       Eating                  85   

         venueType  checkin_cnt checkin_yrwk  weekday  incomce_mean  \
0        Apartment          267       202209        1      0.000000   
1        Apartment          267       202209        1      0.000000   
2        Apartment          267       202209        1      0.000000   
3        Apartment            5       202209        1      0.000000   
4        Apartment            5       202209        1      1.333333   
...            ...          ...          ...      ...           ...   
724402  Restaurant          115       202321        2      5.509499   
724403  Restaurant           39       202321        2      5.082764   
724404  Restaurant           24       202321        2      4.352295   
724405  Restaurant           23       202321        2      5.915035   
724406  Restaurant           85       202321        2      5.593566   

        income_sum  income_max  income_min  
0         0.000000    0.000000    0.000000  
1         0.000000    0.000000    0.000000  
2         0.000000    0.000000    0.000000  
3         0.000000    0.000000    0.000000  
4         4.000000    4.000000    0.000000  
...            ...         ...         ...  
724402  633.592377    5.509499    5.509499  
724403  198.227801    5.082764    5.082764  
724404  104.455073    4.352295    4.352295  
724405  136.045800    5.915035    5.915035  
724406  475.453092    5.593566    5.593566  

[724407 rows x 12 columns]

In [394]:
daily_venue_checkin_income['yr_month'] = daily_venue_checkin_income['checkin_date'].apply(lambda x: x.strftime("%Y-%m"))

In [650]:
daily_venue_checkin_income

checkin_date  venueId                      purpose  checkin_cnt_travel  \
0        2022-03-01        0  Coming Back From Restaurant                 129   
1        2022-03-01        0           Going Back to Home                   7   
2        2022-03-01        0            Work/Home Commute                 131   
3        2022-03-01        1  Coming Back From Restaurant                   1   
4        2022-03-01        1           Going Back to Home                   3   
...             ...      ...                          ...                 ...   
724402   2023-05-24     1801                       Eating                 115   
724403   2023-05-24     1802                       Eating                  39   
724404   2023-05-24     1803                       Eating                  24   
724405   2023-05-24     1804                       Eating                  23   
724406   2023-05-24     1805                       Eating                  85   

         venueType  checkin_cnt checkin_yrwk  weekday  incomce_mean  \
0        Apartment          267       202209        1      0.000000   
1        Apartment          267       202209        1      0.000000   
2        Apartment          267       202209        1      0.000000   
3        Apartment            5       202209        1      0.000000   
4        Apartment            5       202209        1      1.333333   
...            ...          ...          ...      ...           ...   
724402  Restaurant          115       202321        2      5.509499   
724403  Restaurant           39       202321        2      5.082764   
724404  Restaurant           24       202321        2      4.352295   
724405  Restaurant           23       202321        2      5.915035   
724406  Restaurant           85       202321        2      5.593566   

        income_sum  income_max  income_min yr_month  MaxOccupancy  \
0         0.000000    0.000000    0.000000  2022-03         242.0   
1         0.000000    0.000000    0.000000  2022-03         242.0   
2         0.000000    0.000000    0.000000  2022-03         242.0   
3         0.000000    0.000000    0.000000  2022-03           8.0   
4         4.000000    4.000000    0.000000  2022-03           8.0   
...            ...         ...         ...      ...           ...   
724402  633.592377    5.509499    5.509499  2023-05          64.0   
724403  198.227801    5.082764    5.082764  2023-05          72.0   
724404  104.455073    4.352295    4.352295  2023-05          94.0   
724405  136.045800    5.915035    5.915035  2023-05          48.0   
724406  475.453092    5.593566    5.593566  2023-05          85.0   

        Occupancy_ratio  
0              1.103306  
1              1.103306  
2              1.103306  
3              0.625000  
4              0.625000  
...                 ...  
724402         1.796875  
724403         0.541667  
724404         0.255319  
724405         0.479167  
724406         1.000000  

[724407 rows x 15 columns]

In [673]:
monthly_venue_checkin_income = daily_venue_checkin_income[['venueId', 'purpose', 'checkin_cnt_travel', 'venueType', 'yr_month', 'income_sum', 'incomce_mean', 'Occupancy_ratio']].groupby(
    ['venueId', 'purpose', 'venueType', 'yr_month']
).agg(
    monthly_customer_count=('checkin_cnt_travel', np.sum),
    monthly_income=('income_sum', np.sum),
    customer_average_spend=('incomce_mean', np.mean),
    monthly_occupancy_ratio=('Occupancy_ratio', np.sum)
).reset_index()

In [674]:
monthly_venue_checkin_income =  monthly_venue_checkin_income.sort_values(by=['venueId', 'purpose', 'yr_month'])

In [675]:
monthly_venue_checkin_income['last_month_income'] = monthly_venue_checkin_income.groupby(by=['venueId', 'purpose', 'venueType'])['monthly_income'].shift(1)
monthly_venue_checkin_income['last_month_customer_count'] = monthly_venue_checkin_income.groupby(by=['venueId', 'purpose', 'venueType'])['monthly_customer_count'].shift(1)
monthly_venue_checkin_income['last_month_occupancy_ratio'] = monthly_venue_checkin_income.groupby(by=['venueId', 'purpose', 'venueType'])['monthly_occupancy_ratio'].shift(1)
monthly_venue_checkin_income['last_month_customer_average_spend'] = monthly_venue_checkin_income.groupby(by=['venueId', 'purpose', 'venueType'])['customer_average_spend'].shift(1)

In [680]:
monthly_venue_checkin_income['last_month_customer_average_spend'] = monthly_venue_checkin_income['last_month_customer_average_spend'].fillna(0)

In [677]:
monthly_venue_checkin_income.to_csv('monthly_venue_checkin_income.csv')

In [402]:
building_df = pd.read_csv(att_base_path / 'Buildings.csv')

In [410]:
apartment_df = pd.read_csv(att_base_path / 'Apartments.csv')


1

In [424]:
buildings['maxOccupancy'] = buildings['maxOccupancy'].fillna( int(buildings['maxOccupancy'].mean()) )


In [432]:
buildings = buildings[['maxOccupancy', 'units']]
buildings = buildings.dropna()


In [437]:
buildings['units'] = buildings['units'].apply(ast.literal_eval)

C:\Users\TANGGA\AppData\Local\Temp\ipykernel_7988\3123876902.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  buildings['units'] = buildings['units'].apply(ast.literal_eval)


In [439]:
buildings

maxOccupancy                      units
1             12.0  [481, 498, 534, 652, 818]
2             15.0                      [382]
4              2.0                      [231]
7             15.0                     [1339]
9              5.0               [1050, 1233]
...            ...                        ...
1033          15.0                     [1316]
1037           8.0                 [118, 287]
1038           2.0                      [365]
1039           5.0         [1406, 1564, 1569]
1040          15.0                     [1737]

[673 rows x 2 columns]

In [442]:


unit_occupancy = pd.DataFrame([(d, tup.maxOccupancy) for tup in buildings[['maxOccupancy', 'units']].dropna().itertuples() for d in tup.units])

In [443]:
unit_occupancy.columns = ['venueId', 'MaxOccupancy']

In [414]:
# df = pd.DataFrame( {"name" : ["John", "Eric"], 
#                "days" : [[1, 3, 5, 7], [2,4]]})


TypeError: 'float' object is not iterable

In [446]:
daily_venue_checkin_income = daily_venue_checkin_income.merge(unit_occupancy, on=['venueId'], how='left')

In [448]:
daily_venue_checkin_income['Occupancy_ratio'] = daily_venue_checkin_income['checkin_cnt'] /  daily_venue_checkin_income['MaxOccupancy']

In [450]:
daily_venue_checkin_income.to_csv('daily_venue_checkin_income.csv', index=False)

In [453]:
building_df['location'][0]

'POLYGON ((350.0638997002585 4595.665606173783, 390.0632568061789 4595.892390461278, 390.6810201497753 4486.933622371929, 323.6754997155391 4488.923908968658, 324.8980852269488 4595.180890652538, 350.0638997002585 4595.665606173783))'

In [456]:
building_df['localtion']

buildingId                                           location  \
0              1  POLYGON ((350.0638997002585 4595.665606173783,...   
1              2  POLYGON ((-1926.972613718425 2725.610686806701...   
2              3  POLYGON ((685.6846002015491 1552.131491805318,...   
3              4  POLYGON ((-976.7845160060303 4542.38209636188,...   
4              5  POLYGON ((1259.3061988755617 3572.726728111263...   
...          ...                                                ...   
1037        1038  POLYGON ((361.2332515102781 4376.8840047388585...   
1038        1039  POLYGON ((1011.5463728925471 362.9701764035531...   
1039        1040  POLYGON ((-3562.8074734172264 7362.92295511692...   
1040        1041  POLYGON ((-4151.699758553839 5418.613349758049...   
1041        1042  POLYGON ((-1658.924569481752 4827.536640495573...   

     buildingType  maxOccupancy                  units  
0      Commercial           NaN                    NaN  
1      Residental          12.0  [481,498,534,652,818]  
2      Commercial           NaN                  [382]  
3      Commercial           NaN                    NaN  
4      Residental           2.0                  [231]  
...           ...           ...                    ...  
1037   Residental           8.0              [118,287]  
1038   Residental           2.0                  [365]  
1039   Residental           5.0       [1406,1564,1569]  
1040   Commercial           NaN                 [1737]  
1041   Commercial           NaN                    NaN  

[1042 rows x 5 columns]

In [457]:
!pip install django


[notice] A new release of pip is available: 23.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip



     ---------------------------------------- 0.0/8.0 MB ? eta -:--:--
     ---------------------------------------- 0.1/8.0 MB 1.3 MB/s eta 0:00:07
     - -------------------------------------- 0.3/8.0 MB 2.8 MB/s eta 0:00:03
     -- ------------------------------------- 0.5/8.0 MB 4.3 MB/s eta 0:00:02
     ---- ----------------------------------- 0.9/8.0 MB 4.7 MB/s eta 0:00:02
     ----------- ---------------------------- 2.2/8.0 MB 9.4 MB/s eta 0:00:01
     ----------------------- ---------------- 4.7/8.0 MB 16.7 MB/s eta 0:00:01
     ------------------------------- -------- 6.3/8.0 MB 21.1 MB/s eta 0:00:01
     ---------------------------------------  8.0/8.0 MB 23.2 MB/s eta 0:00:01
     ---------------------------------------- 8.0/8.0 MB 20.4 MB/s eta 0:00:00
  Using cached backports.zoneinfo-0.2.1-cp38-cp38-win_amd64.whl (38 kB)
     ---------------------------------------- 0.0/41.2 kB ? eta -:--:--
     ---------------------------------------- 41.2/41.2 kB 1.9 MB/s eta 0:00:0

In [460]:
from django.contrib.gis.geos import Polygon, Point, MultiPoint, GeometryCollection

poly = Polygon( ((0, 0), (0, 1), (1, 1), (0, 0)) )
gc = GeometryCollection(Point(0, 0), MultiPoint(Point(0, 0), Point(1, 1)), poly)
gc.geojson


ImproperlyConfigured: Could not find the GDAL library (tried "gdal306", "gdal305", "gdal304", "gdal303", "gdal302", "gdal301", "gdal300", "gdal204", "gdal203", "gdal202"). Is GDAL installed? If it is, try setting GDAL_LIBRARY_PATH in your settings.

In [459]:
!pip install GDAL

     ---------------------------------------- 0.0/758.1 kB ? eta -:--:--
     ---- ---------------------------------- 81.9/758.1 kB 1.5 MB/s eta 0:00:01
     ---------- --------------------------- 204.8/758.1 kB 2.1 MB/s eta 0:00:01
     ----------------------- -------------- 471.0/758.1 kB 3.3 MB/s eta 0:00:01
     ------------------------------------ - 727.0/758.1 kB 3.8 MB/s eta 0:00:01
     -------------------------------------- 758.1/758.1 kB 3.7 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Running setup.py clean for GDAL
Failed to build GDAL
  Running setup.py install for GDAL: started
  Running setup.py install for GDAL: finished with status 'error'


  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [140 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-38
      creating build\lib.win-amd64-cpython-38\osgeo
      copying osgeo\gdal.py -> build\lib.win-amd64-cpython-38\osgeo
      copying osgeo\gdalconst.py -> build\lib.win-amd64-cpython-38\osgeo
      copying osgeo\gdalnumeric.py -> build\lib.win-amd64-cpython-38\osgeo
      copying osgeo\gdal_array.py -> build\lib.win-amd64-cpython-38\osgeo
      copying osgeo\gnm.py -> build\lib.win-amd64-cpython-38\osgeo
      copying osgeo\ogr.py -> build\lib.win-amd64-cpython-38\osgeo
      copying osgeo\osr.py -> build\lib.win-amd64-cpython-38\osgeo
      copying osgeo\__init__.py -> build\lib.win-amd64-cpython-38\osgeo
      creating build\lib.win-amd64-cpython-38\osgeo_utils
      copying gdal-utils\osgeo_utils\gdal2tiles

In [468]:
building_df['location'][0]

'POLYGON ((350.0638997002585 4595.665606173783, 390.0632568061789 4595.892390461278, 390.6810201497753 4486.933622371929, 323.6754997155391 4488.923908968658, 324.8980852269488 4595.180890652538, 350.0638997002585 4595.665606173783))'

In [465]:
polyDf

X         Y     Name
0  -84.202396  9.480334  Border1
1  -84.202190  9.480555  Border1
2  -84.201886  9.480792  Border1
3  -84.201529  9.481153  Border1
4  -84.201307  9.481303  Border1
5  -84.201234  9.481376  Border1
6  -84.201156  9.481593  Border1
7  -84.201084  9.481722  Border1
8  -84.201029  9.481649  Border1
9  -84.201136  9.481356  Border1
10 -84.201338  9.480802  Border1
11 -84.201015  9.480494  Border1
12 -84.201148  9.480323  Border1
13 -84.201244  9.480264  Border1
14 -84.201298  9.480157  Border1
15 -84.201375  9.479869  Border1
16 -84.201509  9.479760  Border1
17 -84.201635  9.479498  Border1
18 -84.201690  9.479408  Border1
19 -84.201742  9.479385  Border1
20 -84.201863  9.479413  Border1
21 -84.202157  9.479673  Border1
22 -84.202398  9.479917  Border1
23 -84.202396  9.480334  Border1

In [463]:
#import line vertex from csv file
polyDf = pd.read_csv('cropPoly.csv',header=0)
polyDf.head()


#show the last item
polyDf.tail()

# define schema
schema = {
    'geometry':'Polygon',
    'properties':[('Name','str')]
}

#open a fiona object
polyShp = fiona.open('cropPoly.shp', mode='w', driver='ESRI Shapefile', schema = schema, crs = "EPSG:4326")

#get list of points
xyList = []
rowName = ''
for index, row in polyDf.iterrows():
    xyList.append((row.X,row.Y))
    rowName = row.Name
xyList[:5]

#save record and close shapefile
rowDict = {
'geometry' : {'type':'Polygon', 'coordinates': [xyList]}, #Here the xyList is in brackets
'properties': {'Name' : rowName},
}
polyShp.write(rowDict)
#close fiona object
polyShp.close()

In [473]:
import geopandas
from shapely.geometry import Polygon
import shapely.wkt

In [476]:
# shapely.wkt.loads('POLYGON ((51.0 3.0, 51.3 3.61, 51.3 3.0, 51.0 3.0))')

building_df['geometry'] = building_df['location'].apply(shapely.wkt.loads)

In [526]:
polygon['venueId'] = building_df['buildingId']

buildingId                                           location  \
0              1  POLYGON ((350.0638997002585 4595.665606173783,...   
1              2  POLYGON ((-1926.972613718425 2725.610686806701...   
2              3  POLYGON ((685.6846002015491 1552.131491805318,...   
3              4  POLYGON ((-976.7845160060303 4542.38209636188,...   
4              5  POLYGON ((1259.3061988755617 3572.726728111263...   
...          ...                                                ...   
1037        1038  POLYGON ((361.2332515102781 4376.8840047388585...   
1038        1039  POLYGON ((1011.5463728925471 362.9701764035531...   
1039        1040  POLYGON ((-3562.8074734172264 7362.92295511692...   
1040        1041  POLYGON ((-4151.699758553839 5418.613349758049...   
1041        1042  POLYGON ((-1658.924569481752 4827.536640495573...   

     buildingType  maxOccupancy                  units  \
0      Commercial           NaN                    NaN   
1      Residental          12.0  [481,498,534,652,818]   
2      Commercial           NaN                  [382]   
3      Commercial           NaN                    NaN   
4      Residental           2.0                  [231]   
...           ...           ...                    ...   
1037   Residental           8.0              [118,287]   
1038   Residental           2.0                  [365]   
1039   Residental           5.0       [1406,1564,1569]   
1040   Commercial           NaN                 [1737]   
1041   Commercial           NaN                    NaN   

                                               geometry  
0     POLYGON ((350.0638997002585 4595.665606173783,...  
1     POLYGON ((-1926.972613718425 2725.610686806701...  
2     POLYGON ((685.6846002015491 1552.131491805318,...  
3     POLYGON ((-976.7845160060303 4542.38209636188,...  
4     POLYGON ((1259.3061988755617 3572.726728111263...  
...                                                 ...  
1037  POLYGON ((361.2332515102781 4376.8840047388585...  
1038  POLYGON ((1011.5463728925471 362.9701764035531...  
1039  POLYGON ((-3562.8074734172264 7362.92295511692...  
1040  POLYGON ((-4151.699758553839 5418.613349758049...  
1041  POLYGON ((-1658.924569481752 4827.536640495573...  

[1042 rows x 6 columns]

In [479]:

# lat_point_list = [50.854457, 52.518172, 50.072651, 48.853033, 50.854457]
# lon_point_list = [4.377184, 13.407759, 14.435935, 2.349553, 4.377184]

# polygon_geom = Polygon(zip(lon_point_list, lat_point_list))

import geopandas as gpd


polygon.to_file(filename='polygon.geojson', driver='GeoJSON')
polygon.to_file(filename='polygon.gpkg', driver="GPKG")
polygon.to_file(filename='polygon.shp', driver="ESRI Shapefile")


In [534]:
building_df

buildingId                                           location  \
0              1  POLYGON ((350.0638997002585 4595.665606173783,...   
1              2  POLYGON ((-1926.972613718425 2725.610686806701...   
2              3  POLYGON ((685.6846002015491 1552.131491805318,...   
3              4  POLYGON ((-976.7845160060303 4542.38209636188,...   
4              5  POLYGON ((1259.3061988755617 3572.726728111263...   
...          ...                                                ...   
1037        1038  POLYGON ((361.2332515102781 4376.8840047388585...   
1038        1039  POLYGON ((1011.5463728925471 362.9701764035531...   
1039        1040  POLYGON ((-3562.8074734172264 7362.92295511692...   
1040        1041  POLYGON ((-4151.699758553839 5418.613349758049...   
1041        1042  POLYGON ((-1658.924569481752 4827.536640495573...   

     buildingType  maxOccupancy                  units  \
0      Commercial           NaN                    NaN   
1      Residental          12.0  [481,498,534,652,818]   
2      Commercial           NaN                  [382]   
3      Commercial           NaN                    NaN   
4      Residental           2.0                  [231]   
...           ...           ...                    ...   
1037   Residental           8.0              [118,287]   
1038   Residental           2.0                  [365]   
1039   Residental           5.0       [1406,1564,1569]   
1040   Commercial           NaN                 [1737]   
1041   Commercial           NaN                    NaN   

                                               geometry  
0     POLYGON ((350.0638997002585 4595.665606173783,...  
1     POLYGON ((-1926.972613718425 2725.610686806701...  
2     POLYGON ((685.6846002015491 1552.131491805318,...  
3     POLYGON ((-976.7845160060303 4542.38209636188,...  
4     POLYGON ((1259.3061988755617 3572.726728111263...  
...                                                 ...  
1037  POLYGON ((361.2332515102781 4376.8840047388585...  
1038  POLYGON ((1011.5463728925471 362.9701764035531...  
1039  POLYGON ((-3562.8074734172264 7362.92295511692...  
1040  POLYGON ((-4151.699758553839 5418.613349758049...  
1041  POLYGON ((-1658.924569481752 4827.536640495573...  

[1042 rows x 6 columns]

In [538]:
polygon = gpd.GeoDataFrame(index=range(len(building_df['geometry'].tolist())), crs='epsg:4326', geometry=building_df['geometry'].tolist())       
polygon['venueId'] = building_df['buildingId']
polygon['geometry'] = polygon['geometry'].scale(xfact=1/10000, yfact=1/10000, zfact=1.0, origin=(0, 0))
polygon.to_file(filename='scaled_polygon.geojson', driver='GeoJSON')

In [531]:
polygon

geometry  venueId
0     POLYGON ((0.35006 4.59567, 0.39006 4.59589, 0....        1
1     POLYGON ((-1.92697 2.72561, -1.94819 2.73024, ...        2
2     POLYGON ((0.68568 1.55213, 0.64600 1.54713, 0....        3
3     POLYGON ((-0.97678 4.54238, -1.05329 4.53136, ...        4
4     POLYGON ((1.25931 3.57273, 1.29926 3.57475, 1....        5
...                                                 ...      ...
1037  POLYGON ((0.36123 4.37688, 0.39052 4.37672, 0....     1038
1038  POLYGON ((1.01155 0.36297, 1.01117 0.32297, 0....     1039
1039  POLYGON ((-3.56281 7.36292, -3.55734 7.29415, ...     1040
1040  POLYGON ((-4.15170 5.41861, -4.11623 5.41913, ...     1041
1041  POLYGON ((-1.65892 4.82754, -1.72613 4.83063, ...     1042

[1042 rows x 2 columns]

In [543]:
# employer_map

employer_map = gpd.GeoDataFrame(employer_df.drop(columns=['geometry', 'location']), crs='epsg:4326', geometry=employer_df['geometry'])       
# employer_map['venueId'] = building_df['buildingId']
employer_map['geometry'] = employer_map['geometry'].scale(xfact=1/10000, yfact=1/10000, zfact=1.0, origin=(0, 0))
employer_map.to_file(filename='employer_map.geojson', driver='GeoJSON')

In [540]:
employer_df['geometry'] = employer_df['location'].apply(shapely.wkt.loads)


In [541]:
geopandas.GeoDataFrame(
    df, geometry=geopandas.points_from_xy(df.Longitude, df.Latitude))

employerId                                       location  buildingId  \
0           379  POINT (-1849.997168394888 1744.6010147106394)         823   
1           380    POINT (41.51783767879146 418.7264799744545)         154   
2           381   POINT (877.2786575380362 1358.5441805909259)         279   
3           382   POINT (670.3987400004884 1584.4743462106067)           3   
4           383   POINT (829.9556783260775 2163.4803049897623)         146   
..          ...                                            ...         ...   
248        1793  POINT (-3313.211912858075 6185.8337275541235)         738   
249        1794   POINT (-3786.373075235378 4999.485452069126)         802   
250        1795    POINT (-3267.606054814424 6452.56705940432)         953   
251        1796   POINT (-3725.261225376805 6345.551342169127)        1011   
252        1797   POINT (-3759.480918044369 5179.602735764589)         949   

                                          geometry  
0    POINT (-1849.997168394888 1744.6010147106394)  
1      POINT (41.51783767879146 418.7264799744545)  
2     POINT (877.2786575380362 1358.5441805909259)  
3     POINT (670.3987400004884 1584.4743462106067)  
4     POINT (829.9556783260775 2163.4803049897623)  
..                                             ...  
248  POINT (-3313.211912858075 6185.8337275541235)  
249   POINT (-3786.373075235378 4999.485452069126)  
250    POINT (-3267.606054814424 6452.56705940432)  
251   POINT (-3725.261225376805 6345.551342169127)  
252   POINT (-3759.480918044369 5179.602735764589)  

[253 rows x 4 columns]

In [545]:
employer_df

employerId                                       location  buildingId  \
0           379  POINT (-1849.997168394888 1744.6010147106394)         823   
1           380    POINT (41.51783767879146 418.7264799744545)         154   
2           381   POINT (877.2786575380362 1358.5441805909259)         279   
3           382   POINT (670.3987400004884 1584.4743462106067)           3   
4           383   POINT (829.9556783260775 2163.4803049897623)         146   
..          ...                                            ...         ...   
248        1793  POINT (-3313.211912858075 6185.8337275541235)         738   
249        1794   POINT (-3786.373075235378 4999.485452069126)         802   
250        1795    POINT (-3267.606054814424 6452.56705940432)         953   
251        1796   POINT (-3725.261225376805 6345.551342169127)        1011   
252        1797   POINT (-3759.480918044369 5179.602735764589)         949   

                                          geometry  
0    POINT (-1849.997168394888 1744.6010147106394)  
1      POINT (41.51783767879146 418.7264799744545)  
2     POINT (877.2786575380362 1358.5441805909259)  
3     POINT (670.3987400004884 1584.4743462106067)  
4     POINT (829.9556783260775 2163.4803049897623)  
..                                             ...  
248  POINT (-3313.211912858075 6185.8337275541235)  
249   POINT (-3786.373075235378 4999.485452069126)  
250    POINT (-3267.606054814424 6452.56705940432)  
251   POINT (-3725.261225376805 6345.551342169127)  
252   POINT (-3759.480918044369 5179.602735764589)  

[253 rows x 4 columns]

In [546]:
raw_file

jobId  employerId  hourlyRate           startTime             endTime  \
0         0         379   10.000000 1900-01-01 07:46:00 1900-01-01 03:46:00   
1         1         379   22.217633 1900-01-01 07:31:00 1900-01-01 03:31:00   
2         2         380   10.000000 1900-01-01 08:00:00 1900-01-01 04:00:00   
3         3         380   15.312071 1900-01-01 07:39:00 1900-01-01 03:39:00   
4         4         381   21.355409 1900-01-01 07:53:00 1900-01-01 03:53:00   
...     ...         ...         ...                 ...                 ...   
1323   1323        1796   37.166553 1900-01-01 07:29:00 1900-01-01 03:29:00   
1324   1324        1796   23.815211 1900-01-01 08:14:00 1900-01-01 04:14:00   
1325   1325        1797   13.117569 1900-01-01 07:37:00 1900-01-01 03:37:00   
1326   1326        1797   10.000000 1900-01-01 07:31:00 1900-01-01 03:31:00   
1327   1327        1797   11.932836 1900-01-01 08:19:00 1900-01-01 04:19:00   

                                       daysToWork educationRequirement  \
0      [Monday,Tuesday,Wednesday,Thursday,Friday]  HighSchoolOrCollege   
1      [Monday,Tuesday,Wednesday,Thursday,Friday]            Bachelors   
2      [Monday,Tuesday,Wednesday,Thursday,Friday]            Bachelors   
3      [Monday,Tuesday,Wednesday,Thursday,Friday]            Bachelors   
4      [Monday,Tuesday,Wednesday,Thursday,Friday]  HighSchoolOrCollege   
...                                           ...                  ...   
1323   [Monday,Tuesday,Wednesday,Thursday,Friday]             Graduate   
1324  [Friday,Saturday,Thursday,Monday,Wednesday]            Bachelors   
1325   [Monday,Tuesday,Wednesday,Thursday,Friday]  HighSchoolOrCollege   
1326   [Monday,Tuesday,Wednesday,Thursday,Friday]  HighSchoolOrCollege   
1327   [Monday,Tuesday,Wednesday,Thursday,Friday]            Bachelors   

      days_per_week  monthly_salary  
0                 5     1600.000000  
1                 5     3554.821338  
2                 5     1600.000000  
3                 5     2449.931302  
4                 5     3416.865486  
...             ...             ...  
1323              5     5946.648483  
1324              5     3810.433757  
1325              5     2098.811030  
1326              5     1600.000000  
1327              5     1909.253723  

[1328 rows x 9 columns]

In [564]:
pd.to_datetime('7:29:00 PM', format='%I:%M:%S %p')

Timestamp('1900-01-01 19:29:00')

In [648]:
job_df = pd.read_csv(att_base_path / 'Jobs.csv')
job_df['startTime'] = pd.to_datetime(job_df['startTime'], format='%I:%M:%S %p')
job_df['endTime'] = pd.to_datetime(job_df['endTime'], format='%I:%M:%S %p')
job_df['workPerWeek'] = job_df['daysToWork'].apply(lambda x: len(x.split(',')))
job_df['workHour'] = (job_df['endTime']-job_df['startTime']) /  np.timedelta64(1,'h')
education_map = dict(zip(set(job_df['educationRequirement']), [4,3,2,1]))
job_df['jobEducationLevel'] = job_df['educationRequirement'].apply(lambda x: education_map[x])
job_df = job_df.drop(columns=['educationRequirement', 'startTime', 'endTime', 'daysToWork'])
participant_df = pd.read_csv(att_base_path / 'participants.csv')
participant_df['partEducationLevel'] = participant_df['educationLevel'].apply(lambda x: education_map[x])
participant_df = participant_df.drop(columns ='educationLevel')
pid_jid_daily = raw_file[raw_file['currentMode'] == 'AtHome'][['participantId', 'date', 'jobId', 'apartmentId']].drop_duplicates()
print(pid_jid_daily.shape)
pid_jid_daily = pid_jid_daily.merge(job_df, on ='jobId', how='left')
print(pid_jid_daily.shape)
pid_jid_daily = pid_jid_daily.merge(participant_df, on ='participantId', how='left')
print(pid_jid_daily.shape)
pid_jid_daily['educationMatchRatio'] = pid_jid_daily['partEducationLevel'] / pid_jid_daily['jobEducationLevel'] 
print(pid_jid_daily.shape)
apartment_df = pd.read_csv(att_base_path / 'Apartments.csv')
apartment_df['geometry'] = apartment_df['location'].apply(shapely.wkt.loads)
apartment_df = apartment_df.drop(columns=['location', 'rentalCost'])
pid_jid_daily = pid_jid_daily.merge(apartment_df, on ='apartmentId', how='left')
print(pid_jid_daily.shape)
pid_jid_daily['date'] = pid_jid_daily['date'].apply(lambda x: x.strftime('%Y-%m-%d'))

pid_jid_daily_detail_map = gpd.GeoDataFrame(pid_jid_daily.drop(columns=['geometry']), crs='epsg:4326', geometry=pid_jid_daily['geometry'])       
pid_jid_daily_detail_map['geometry'] = pid_jid_daily_detail_map['geometry'].scale(xfact=1/10000, yfact=1/10000, zfact=1.0, origin=(0, 0))
pid_jid_daily_detail_map.to_file(filename='pid_jid_daily_detail_map_v2.geojson', driver='GeoJSON')

(398585, 4)
(398585, 9)
(398585, 15)
(398585, 16)
(398585, 20)


In [649]:

pid_jid_daily_detail_map = gpd.GeoDataFrame(pid_jid_daily.drop(columns=['geometry']), crs='epsg:4326', geometry=pid_jid_daily['geometry'])       
pid_jid_daily_detail_map['geometry'] = pid_jid_daily_detail_map['geometry'].scale(xfact=1/10000, yfact=1/10000, zfact=1.0, origin=(0, 0))
pid_jid_daily_detail_map.to_file(filename='pid_jid_daily_detail_map_v2.geojson', driver='GeoJSON')

In [642]:
participant_home = raw_file[raw_file['currentMode'] == 'AtHome'][['participantId', 'apartmentId']].drop_duplicates()

In [644]:
participant_home.groupby(by=['participantId']).size().reset_index().sort_values(0)

participantId  0
0              0.0  1
663          663.0  1
664          664.0  1
665          665.0  1
666          666.0  1
..             ... ..
798          798.0  3
559          559.0  3
576          576.0  3
141          141.0  3
372          372.0  4

[1011 rows x 2 columns]

In [647]:
participant_home[participant_home['participantId'] == 372]

participantId  apartmentId
372             372.0          NaN
496773          372.0        784.0
554400          372.0        732.0
804117          372.0       1010.0